public/app/plugins/panel/table/transformers.ts

```diff
@@ -73,6 +73,56 @@ transformers['timeseries_to_columns'] = {
   }
 };
 
+transformers['grouping_to_rows_columns'] = {
+  description: 'Grouping to rows and columns',
+  getColumns: function() {
+    return [];
+  },
+  transform: function(data, panel, model) {
+    // Getting the columns
+    const points = new Map<string, Map<string, any>>();
+
+    const columns = new Set<string>();
+    const rows = new Set<string>();
+
+    for (let series of data) {
+      const columnsString = series.target.slice(
+        series.target.indexOf("{") + 1, -1);
+      const columnsArray = columnsString.split(", ");
+      const column = columnsArray[0].split(": ")[1];
+      const row = columnsArray[1].split(": ")[1];
```

Comment:
Until this point, the code is figuring out what are the groupings that are being made in the query. I have access only to the string, so this part is quite tricky. Can you give me any hint about how to get the groupings of the query in a programmatic way?